LanguageFeatures/Patterns/type_inference_A08_t03.dart

```diff
@@ -51,10 +53,16 @@ main() {
     v2.expectStaticType<Exactly<B>>();
     v3.expectStaticType<Exactly<C>>();
   }
+  String log = "";
   try {
-    final {"key1": A v1, "key2": B v2, "key3": v3} = {};
-    v1.expectStaticType<Exactly<A>>();
-    v2.expectStaticType<Exactly<B>>();
-    v3.expectStaticType<Exactly<B>>();
+    var {"key1": A v1, "key2": B v2, "key3": v3} =
+        getType({}, (String s) {log += s;});
+  } catch (_) {}
+  Expect.equals("Map<Object?, B>", log);
```

Comment:
I think `getType` is invoked here with a context type of `Map<_, B>`, so the result may well be `Map<dynamic, B>`.